### ISSUE TITLE
Random sneaking.

### ISSUE THREAD

**Issue by Drogobo** [MeteorDevelopment/meteor-client#3113]
### Describe the bug

I am randomly sneaking when I have no modules enabled that could cause that. I only have render modules enabled. The random sneaking only lasts for like 1 game tick, but it vexes me so much. The slight imperfection makes me want to throw up. Here are the modules I have enabled:
- Better Tooltips
- Fullbright
- No Render
- Unfocused Cpu
- Waypoints
I have tested Minecraft without using Meteor, and it seems to go away. I am experiencing this on single player hardcore worlds.

### Steps to reproduce

```bash
Load up Meteor Client on 1.19.3
Enable the listed modules
Wait a little bit
WTF!!! Random crouching!
```

### Meteor Version

meteor-client-0.5.2-1741

### Minecraft Version

1.19.3

### Operating System

Linux

### Before submitting a bug report

- [x] This bug wasn't already reported (I have searched bug reports on GitHub).

- [X] This is a valid bug (I am able to reproduce this on the latest dev build).

**Comment by I-Got-You-Dead** [comment]
Keyboard issue

**Comment by Drogobo** [comment]
> Keyboard issue

I can assure you, it happens in crafting tables and chests. If my keyboard was randomly pressing keys, then ok, but my keyboard can't make me sneak while in menus.

**Comment by Wide-Cat** [comment]
are you sure that these are the only modules you have enabled? Also, what other mods are you using?

**Comment by Drogobo** [comment]
> are you sure that these are the only modules you have enabled? Also, what other mods are you using?

I am 100% it is Meteor. Here are other mods I am using with Meteor:
- BiomeThreadLocalFix
- Fabric API
- Krypton
- LazyDFU
- Sodium
- Starlight

All of these mods are approved by the java speedrunning team. These are not the culprit.

**Comment by Drogobo** [comment]
https://youtu.be/BA3Ctk1A4Ko
here is a video I got showcasing the bug

**Comment by qeqeqeplayer533** [comment]
that sometimes happen to me as well, but not an issieu

**Comment by Drogobo** [comment]
I have compiled meteor from source, and it seems to have gone away. Weird that it does that.

**Comment by Drogobo** [comment]
Nvm, it is happening again. I wonder which commit caused this because I did not experience this before.

**Comment by Drogobo** [comment]
I am at commit 70f1a30c661d97da4ac11ce658e3f036bf62f052, and the issue is not present. I compiled 70f1a30c661d97da4ac11ce658e3f036bf62f052 from source and it does not have this bug. https://github.com/MeteorDevelopment/meteor-client/commit/70f1a30c661d97da4ac11ce658e3f036bf62f052

**Comment by I-Got-You-Dead** [comment]
So a few days ago I booted up a fresh instance of meteor 1.19.3 and had the same issue

**Comment by I-Got-You-Dead** [comment]
I can confirm this is a problem with meteor

**Comment by Drogobo** [comment]
> I can confirm this is a problem with meteor. (Could be fabric or minecraft itself, although unlikely)

Nope. as I said, I compiled from the commit https://github.com/MeteorDevelopment/meteor-client/commit/70f1a30c661d97da4ac11ce658e3f036bf62f052, and it worked.

**Comment by I-Got-You-Dead** [comment]
Very interesting

**Comment by ThebestkillerTBK** [comment]
guess its baritone issue

**Comment by ThebestkillerTBK** [comment]
guess its baritone issue

**Comment by AnhNguyenlost13** [comment]
i dont think baritone interferes with movement when inactive

**Comment by ThebestkillerTBK** [comment]
> i dont think baritone interferes with movement when inactive

meteor doent interfere input tho

**Comment by I-Got-You-Dead** [comment]
Motor*

**Comment by Drogobo** [comment]
This is 100% meteor client. It stops when I disable meteor.

**Comment by ThebestkillerTBK** [comment]
today the game start to press directional keys randomly :skull:

**Comment by ThebestkillerTBK** [comment]
> This is 100% meteor client. It stops when I disable meteor.

I still cannot find the issue inside meteor 😭

**Comment by I-Got-You-Dead** [comment]
> > This is 100% meteor client. It stops when I disable meteor.
> 
> I still cannot find the issue inside meteor 😭

I would assume it’s maybe mixin related

**Comment by ThebestkillerTBK** [comment]
> 

i can find no clue in the mixins

**Comment by I-Got-You-Dead** [comment]
What if you were to create a new branch and just remove the keyboard mixin and anything related

**Comment by I-Got-You-Dead** [comment]
If it stops we know it’s that

**Comment by ThebestkillerTBK** [comment]
the problem seems to be gone after removing KeyboardInputMixin and KeyboardMixin

**Comment by I-Got-You-Dead** [comment]
Perfect

**Comment by Drogobo** [comment]
a PR that was recently made fixes this issue. I will close this if it gets merged, and I hope it does.

**Comment by Drogobo** [comment]
PR was merged! GG this issue is no more.

**Comment by Drogobo** [comment]
Reopening because this is an issue on the newest dev builds

**Comment by I-Got-You-Dead2** [comment]
It’s a vanilla issue

**Comment by Drogobo** [comment]
> It’s a vanilla issue

nope, I turned meteor off and and it is still doing it's thing

**Comment by I-Got-You-Dead2** [comment]
May I ask what are your exact pc specs

**Comment by I-Got-You-Dead2** [comment]
I have a feeling it’s related to fps or something like that

**Comment by Drogobo** [comment]
> I have a feeling it’s related to fps or something like that

I already said, it is 100% a meteor thing. Stop denying it.

**Comment by Drogobo** [comment]
I played Minecraft for a few hours without meteor. When I turned it on, the issue returned.

**Comment by I-Got-You-Dead2** [comment]
That’s nice

**Comment by I-Got-You-Dead2** [comment]
Cope

**Comment by Drogobo** [comment]
Is anyone gonna get on this? This is horrible to play with.

**Comment by I-Got-You-Dead2** [comment]
No

**Comment by I-Got-You-Dead2** [comment]
Because for the last time it’s not a meteor issue

**Comment by Drogobo** [comment]
> Because for the last time it’s not a meteor issue

Well, it is a meteor issue. I turned it off and it stops, but if I turn it on it starts right back up.

**Comment by I-Got-You-Dead2** [comment]
Well that’s a fucking you problem

**Comment by Drogobo** [comment]
> Well that’s a fucking you problem

You don't even work on meteor, why are you in this bug report to tell me it is an issue on my end?

**Comment by I-Got-You-Dead2** [comment]
“Contributor”

**Comment by Akul2010** [comment]
Jeez calm down. I'm having the same problem and saw this issue.

**Comment by Starmania** [comment]
Ok, does an alt or a friend see you crouching too ? (visual bug ?)

**Comment by Akul2010** [comment]
I don't know, I normally play alone. As @Drogobo said, it happened only with Meteor, though. I have no other mods besides this.

**Comment by Drogobo** [comment]
What I can say is that it will sometimes happen in the inventory. I can see the my character model do a crouch for 1 second in the inventory GUI.

**Comment by MineGame159** [comment]
Fixed in e2854ceb11c621c45d481680af8559c14b992baa.
